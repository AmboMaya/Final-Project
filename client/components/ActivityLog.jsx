import React from 'react'
import { Button, Divider, Form, TextArea, Modal, Card } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { addLog } from '../actions/records'

class ActivityLog extends React.Component {
  state = {
    open: false,
    log: ""
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  submitHandler = e => {
    e.preventDefault()
    this.props.addLog(this.props.user, {
      activityId: this.props.id,
      log: this.state.log
    })
  }

  handleRef = component => (this.ref = component)
  open = () => this.setState({ open: true }, () => this.ref.focus())
  close = () => this.setState({ open: false })
  render() {
    return (
      <div>
        <a size="mini" onClick={this.open}>
          <i className="plus icon" />
          <span>Add log</span>
        </a>
        <Modal
          className="addLogModal"
          size="mini"
          open={this.state.open}
          onClose={this.close}
          closeIcon
        >
          <Modal.Content>
            <Card.Content align="center" />
            <Form onSubmit={this.submitHandler}>
              <TextArea
                name="log"
                value={this.state.log}
                onChange={this.handleChange}
                ref={this.handleRef}
                style={{ minWidth: 250 }}
              />
              <Divider />
              <Button
                type="submit"
                basic={true}
                size="mini"
                onClick={this.close}
              >
                Add
              </Button>
            </Form>
          </Modal.Content>
        </Modal>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    user: state.user
  }
}

const mapDispatchToProps = dispatch => {
  return {
    addLog: (id, record) => dispatch(addLog(id, record))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ActivityLog)
