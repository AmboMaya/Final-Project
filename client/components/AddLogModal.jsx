// import React from 'react'
// import { Placeholder, Form, Modal } from 'semantic-ui-react'

// class AddLogModal extends Component {
//   state = {  }
//   render() { 
//     return ( 
//       <Modal size={size} open={open} onClose={this.close}>
//         <Modal.Content>
//           <Form>
//             <Placeholder>Add Log</Placeholder>
//           </Form>
//         </Modal.Content>
//         <Modal.Actions>
//           <Button positive icon='checkmark' labelPosition='right' content='Yes' />
//         </Modal.Actions>
//       </Modal>
//      )
//   }
// }
 
// export default AddLogModal