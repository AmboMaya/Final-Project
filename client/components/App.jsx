import React from 'react'

class App extends React.Component {
  render () {
    return (
      <Main />
    )
  }
}

export default App
