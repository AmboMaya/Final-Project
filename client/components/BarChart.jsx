import React, {Component} from 'react'
import {Line} from 'react-chartjs-2'
import {Card, Grid, Modal} from 'semantic-ui-react'

export default class BarChart extends Component {
  render () {
    return (
      <React.Fragment>
        <Card fluid>
          <Card.Content align="center">
            <Card.Header>Average Ratings</Card.Header>
          </Card.Content >
          <Card.Content align="center" >
            <Line
              data={this.props.chartData}
              width={100}
              height={80}
              options={{
                maintainAspectRatio: true,
                legend: {
                  display: true,
                  position: 'bottom',
                  labels: {
                    boxWidth: 30,
                    fontColor: 'black'
                  }
                },
                scales: {
                  yAxes: [{
                    gridLines: {
                      display: false
                    },
                    ticks: {
                      fontColor: 'black',
                      callback: function (label, index, labels) {
                        return yLabels[label]
                      }
                    }
                  }],
                  xAxes: [{
                    gridLines: {
                      display: false
                    },
                    ticks: {
                      fontColor: 'black'
                    }
                  }]
                }
              }}
            />
          </Card.Content>
        </Card>
      </React.Fragment>
    )
  }
}

let yLabels = {
  // 1: 'terrible',
  // 2: 'bad',
  // 3: 'average',
  // 4: 'good',
  // 5: 'great'
  1: 1, 2: 2, 3: 3, 4: 4, 5: 5
}
