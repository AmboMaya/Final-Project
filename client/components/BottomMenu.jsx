import React from 'react'


const BottomMenu = () => {
  return (
    <div className="ui teal three item inverted menu">
      <a className="active item" path='/'>
    Home
      </a>
      <a className="item">
    Calendar
      </a>
      <a className="item">
    Graph
      </a>
    </div>
  )
}

export default BottomMenu
