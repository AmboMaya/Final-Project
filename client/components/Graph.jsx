import React, {Component} from 'react'
import {Line} from 'react-chartjs-2'
import {Card} from 'semantic-ui-react'
import windowSize from 'react-window-size'

class Graph extends Component {
  getGraphHeight () {
    if (this.props.windowWidth < 500) {
      return 90
    } else {
      return 60
    }
  }

  render () {
    return (
      <React.Fragment>
        <Card fluid>
          <Card.Content align="center">
            <Card.Header>Progress graph</Card.Header>
          </Card.Content >
          <Card.Content align="center" className='graphHeader'>
            <Line
              data={this.props.chartData}
              width={100}
              height={this.getGraphHeight}
              options={{
                maintainAspectRatio: true,
                legend: {
                  display: true,
                  position: 'bottom',
                  labels: {
                    boxWidth: 30,
                    fontColor: 'black'
                  }
                },
                scales: {
                  yAxes: [{
                    gridLines: {
                      display: false
                    },
                    ticks: {
                      fontColor: 'black',
                      callback: function (label, index, labels) {
                        return yLabels[label]
                      }
                    }
                  }],
                  xAxes: [{
                    gridLines: {
                      display: false
                    },
                    ticks: {
                      fontColor: 'black'
                    }
                  }]
                }
              }}
            />
          </Card.Content>
        </Card>
      </React.Fragment>
    )
  }
}

let yLabels = {
  // 1: 'terrible',
  // 2: 'bad',
  // 3: 'average',
  // 4: 'good',
  // 5: 'great'
  1: 1, 2: 2, 3: 3, 4: 4, 5: 5
}

export default windowSize(Graph)
