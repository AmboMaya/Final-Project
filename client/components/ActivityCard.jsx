import React from 'react'
import {Card, Grid, Modal} from 'semantic-ui-react'
import Smiley from './Rating/Smiley'
import ActivityLog from './ActivityLog'

const ActivityCard = props => {
  let smilies = []
  for (let i = 1; i <= 5; i++) {
    smilies.push(<Smiley key={i} rating={i} selected={props.currentRating === i} activity={props.activity} />)
  }

  return (
    <React.Fragment>
      <Grid.Column align='center'>
        <Card>
          <Card.Content >
            <Grid>
              <Grid.Column floated='right' width={5} style={{paddingLeft: 35}}>
                <Modal trigger={<a><i className='question circle outline icon right'></i></a>} closeIcon>
                  <Modal.Header>{props.name}</Modal.Header>
                  <Modal.Content>
                    <Modal.Description>
                      <p>{props.info}</p>
                      {props.link ? <a href={props.link}>Source</a> : null}
                    </Modal.Description>
                  </Modal.Content>
                </Modal>
              </Grid.Column>
            </Grid>
            <Card.Header >{props.name}</Card.Header>
            {smilies}
          </Card.Content>
          <Card.Content extra>
            <ActivityLog />
          </Card.Content>
        </Card>
      </Grid.Column>
    </React.Fragment>
  )
}

export default ActivityCard
