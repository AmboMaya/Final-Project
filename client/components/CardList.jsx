import React from 'react'
import { connect } from 'react-redux'

import { Container, Grid } from 'semantic-ui-react'
import ActivityCard from './ActivityCard'
import { getActivities } from '../actions/journalActions'

import BottomMenu from './BottomMenu'

class CardList extends React.Component {
  state = {
    smiles: [
      { mood: 'fa-angry', value: 1 },
      { mood: 'fa-frown', value: 2 },
      { mood: 'fa-meh', value: 3 },
      { mood: 'fa-smile', value: 4 },
      { mood: 'fa-laugh', value: 5 }
    ],
    records: [
      {
        user_id: '1',
        entries: []
      }
    ]
  }

  addRecord = (id, record) => {
    const newRecord = this.state.records.map(rec => {
      if (rec.user_id !== id) return rec
      return {
        ...rec,
        records: [...rec.entries, record]
      }
    })
    this.setState({ records: newRecord })
  }


  componentDidMount() {
    this.props.getActivities()
  }

  render() {
    return (
      <React.Fragment>
        <Container className="appBody">
          <Grid columns={3} doubling stackable>
            {this.props.activities.map((act, key) => {
              return (
                <ActivityCard
                  key={key}
                  name={act.name}
                  act_id={act.id}
                  user_id={this.state.records[0].user_id}
                  smiles={this.state.smiles}
                  addRecord={this.addRecord}
                />
              )
            })}
          </Grid>
        </Container>
        <BottomMenu submit={this.handleSubmit} />
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => {
  const activities = [...state.activities]
  return {
    activities
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getActivities: () => dispatch(getActivities()),
    addNewRecord: newRecord => dispatch(addNewRecord(newRecord))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CardList)
