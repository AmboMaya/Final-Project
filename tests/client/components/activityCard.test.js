import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'


import {ActivityCard} from '../../../client/components/ActivityCard'

configure({adapter: new Adapter()})

test('<ActivityCard /> matches the last snapshot', () => {
  const renderer = new ShallowRenderer()
  const actual = renderer.render(<ActivityCard smiles={[]}/>)
  expect(actual).toMatchSnapshot()
})

