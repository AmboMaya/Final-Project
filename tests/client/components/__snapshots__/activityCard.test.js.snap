// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityCard /> matches the last snapshot 1`] = `
<React.Fragment>
  <GridColumn
    align="center"
  >
    <Card>
      <CardContent
        align="center"
      >
        <Grid>
          <GridColumn
            floated="right"
            width={5}
          >
            <Modal
              centered={true}
              closeIcon={true}
              closeOnDimmerClick={true}
              closeOnDocumentClick={false}
              dimmer={true}
              eventPool="Modal"
              trigger={
                <a>
                  <i
                    className="question circle outline icon right"
                  />
                </a>
              }
            >
              <ModalHeader />
              <ModalContent>
                <ModalDescription>
                  <p />
                </ModalDescription>
              </ModalContent>
            </Modal>
          </GridColumn>
        </Grid>
        <CardHeader />
        <GridColumn
          align="center"
        />
      </CardContent>
      <CardContent
        extra={true}
      >
        <Connect(ActivityLog)
          icon="plus icon"
          text="Add Log"
        />
      </CardContent>
    </Card>
  </GridColumn>
</React.Fragment>
`;
